docs/content/guides/dagster-pipes/kubernetes.mdx

```diff
@@ -0,0 +1,243 @@
+  return k8s_pipes_client.run(
+      context=context,
+      image="pipes-materialize:latest",
+      kubeconfig_file=os.path.expanduser("~/.kube/config"),
+      kube_context="<YOUR_CONTEXT_NAME>"
+  ).get_materialize_result()
+
+
+defs = Definitions(
+  assets=[k8s_pipes_asset],
+  resources={
+    "k8s_pipes_client": PipesK8sClient(),
+  },
+)
+```
+
+---
+
+## Step 2: Write a script for execution on Kubernetes
+
+The next step is to write the code that will be executed on Kubernetes.
+
+TODO
+
+```python
+# my_python_script.py
+
+from dagster_pipes import open_dagster_pipes
+
+with open_dagster_pipes() as pipes:
+    # Stream log message back to Dagster
+    pipes.log.info(f"Using some_parameter value: {some_parameter_value}")
+
+    # ... your code that computes and persists the asset
+
+    # Stream asset materialization metadata and data version back to Dagster.
+    # This should be called after you've computed and stored the asset value. We
+    # omit the asset key here because there is only one asset in scope, but for
+    # multi-assets you can pass an `asset_key` parameter.
+    pipes.report_asset_materialization(
+        metadata={
+            "some_metric": {"raw_value": some_parameter_value + 1, "type": "int"}
+        },
+        data_version="alpha",
+    )
+```
+
+Let's review what this code does:
+
+- Imports <PyObject object="open_dagster_pipes" module="dagster_pipes" /> from `dagster_pipes`
+
+- **Initializes the Dagster Pipes context (<PyObject object="open_dagster_pipes" module="dagster_pipes" />), which yields an instance of <PyObject object="PipesContext" module="dagster_pipes" /> called `pipes`.**
+
+  We're using the default context loader (<PyObject object="PipesDefaultContextLoader" module="dagster_pipes" />) and message writer (<PyObject object="PipesDefaultMessageWriter" module="dagster_pipes" />) in this example. These objects establish communication between the orchestration and external process. On the orchestration end, these match a corresponding `PipesContextInjector` and `PipesMessageReader`, which are instantiated inside the <PyObject module="dagster_k8s" object="PipesK8sClient" />.
+
+- **Inside the body of the context manager (<PyObject object="open_dagster_pipes" module="dagster_pipes" />), retrieve a log and report an asset materialization.** These calls use the temporary communications channels established by <PyObject object="PipesDefaultContextLoader" module="dagster_pipes" /> and <PyObject object="PipesDefaultMessageWriter" module="dagster_pipes" />. To see the full range of what you can do with the <PyObject object="PipesContext" module="dagster_pipes" />, see the API docs or the general [Pipes documentation](/guides/dagster-pipes).
+
+At this point you can execute the rest of your Kubernetes code as normal, invoking various <PyObject object="PipesContext" module="dagster_pipes" /> APIs as needed.
+
+#### Existing codebases
```

Comment:
Pulled this from the updated Databricks guide - I assume it's also applicable here, but will remove / update as needed.

Reply:
unclear to me that this information belongs in a deployment specific guide(s) versus some general pipes place

Reply:
I hear ya; I'm torn on this too. I think it could be useful for visibility and optimizing for less clicking around. Perhaps a side-by-side of new vs existing code, like the following, could work? This info seems generally applicable across Pipes and would be easy to single-source/re-use.

![Screenshot 2023-11-08 at 1 18 23 PM](https://github.com/dagster-io/dagster/assets/16615212/f963b3de-9f7a-4c00-bf9b-b8ae6e611d8b)